doc/route_stop_pattern.md

```diff
@@ -0,0 +1,278 @@
+# Transitland RouteStopPattern
+
+Transitland models route geometries by breaking them into individual components called Route Stop Patterns. These components are uniquely defined by a route, a stop pattern, and a line geometry; all three derived from the trip routes, trip stop sequences, and shapes of a GTFS feed. Because of this, it is possible to have two distinct Route Stop Patterns within one route, both sharing the same line geometry but having different stop patterns, and vice versa. Individual Route Stop Patterns also have records of the GTFS trips and the single shape used to create them; a typical Route Stop Pattern will reference back to one or many trips having the same stop pattern, but only references the one shape shared by those trips. When a Route Stop Pattern's trips have no shapes or empty shapes, there will be no shape reference.
```

Comment:
For the sake of consistency, I'd suggest always calling it a `RouteStopPattern` (without spaces and surrounded by back ticks to format it as a bit of code).